src/librustdoc/clean/auto_trait.rs

```diff
@@ -723,8 +723,8 @@ impl<'a, 'tcx> AutoTraitFinder<'a, 'tcx> {
 
     fn is_fn_ty(&self, ty: &Type) -> bool {
         let tcx = self.cx.tcx;
-        match ty {
-            &Type::ResolvedPath { did, .. } => {
+        match *ty {
+            Type::ResolvedPath { did, .. } => {
```

Comment:
What benefit does this change have? It just seems like churn.

Reply:
In case we add other pattern matching, it allows to not have `&` (which isn't very useful to change in this context, I agree). Oh well, clippy lints. :)

I found this one interesting for future developments so I decided to "fix" it.